### PR TITLE
VEIOSS-241 Allow Excluded Traders to Re-join the IOSS Scheme

### PR DESCRIPTION
**Pull request by pbyrne84**

### CODE REVIEW

File: app/models/etmp/EtmpDisplayRegistration.scala [hunk 1]
```diff
@@ -18,14 +18,42 @@ package models.etmp
 
 import play.api.libs.json.{Json, OFormat}
 
+import java.time.LocalDate
+
 case class EtmpDisplayRegistration(
                                     tradingNames: Seq[EtmpTradingName],
```
Comment (pbyrne84) [inline app/models/etmp/EtmpDisplayRegistration.scala:24]: Logic copied from frontend, so could be wrong in 2 places

File: app/views/YourAccountView.scala.html [hunk 2]
```diff
@@ -30,6 +30,7 @@
         iossNumber: String,
         paymentsViewModel: PaymentsViewModel,
         changeYourRegistrationUrl: String,
+  rejoinRegistrationUrl: Option[String],
```
Comment (hmrcvmu) [inline app/views/YourAccountView.scala.html:33]: indent

File: test/models/etmp/EtmpDisplayRegistrationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        NoLongerSupplies,
+        CeasedTrade,
+        NoLongerMeetsConditions,
+        FailsToComply,
+        VoluntarilyLeaves,
+        TransferringMSID
+      )
+
+      "return true" - {
+        "when it is not a reversal, not quarantined, not quarantined and the effective date is not in the future" in {
```
Comment (hmrcvmu) [inline test/models/etmp/EtmpDisplayRegistrationSpec.scala:69]: `not quarantined, not quarantined and` -> `not quarantined and` ?

Comment (pbyrne84) [reply test/models/etmp/EtmpDisplayRegistrationSpec.scala]: Fixed in other frontend as well

        "when it is not a reversal, not quarantined and the effective date is not in the future"

**Pull request merged**